src/components/modal/__internal__/modal-manager.js

```diff
@@ -1,5 +1,12 @@
 class ModalManagerInstance {
-  #modalList = [];
+  #modalList;
+
+  constructor() {
+    // Due to possibility of multiple carbon versions using it
+    // it is necessary to maintain same structure in this global variable
+    window.__CARBON_INTERNALS_MODAL_LIST = [];
```

Comment:
I think you should check if this global variable already exists. That would be necessary to cover a scenario where one modal opens (first ModalManagerInstance is created) and then content of the modal is being lazy loaded, and second ModalManagerInstance is created.

Reply:
Yeah I had removed it unintentionally, added back in thanks